File: armod_decryptor.py
import base64
from Crypto.Cipher import AES
import json
import re
from urllib.parse import unquote

ENCRYPTION_KEY = "YXJ0dW5uZWw3ODc5Nzg5eA=="

def decrypt_config(encrypted_config):
    key = base64.b64decode(ENCRYPTION_KEY)
    cipher = AES.new(key, AES.MODE_ECB)
    decoded_value = base64.b64decode(encrypted_config)
    decrypted_value = cipher.decrypt(decoded_value)
    return decrypted_value.decode('utf-8')

def process_config(config):
    pattern_vmess = "^ar-(vmess)://"
    pattern_ssh = "^ar-(ssh|vless|socks|trojan-go|trojan|ssr)://"

    matcher_vmess = re.search(pattern_vmess, config)
    matcher_ssh = re.search(pattern_ssh, config)

    if matcher_vmess:
        encrypted_config = config[matcher_vmess.end():]
        decrypted_config = decrypt_config(encrypted_config)
        decoded_config = unquote(decrypted_config)
        return format_decryption_result("VMess", decoded_config)
    elif matcher_ssh:
        encrypted_config = config[matcher_ssh.end():]
        decrypted_config = decrypt_config(encrypted_config)
        decoded_config = unquote(decrypted_config)
        decoded_config = replace_needless_characters(decoded_config)
        return format_decryption_result("SSH", decoded_config)
    else:
        return "Invalid config or unlock config"

def format_decryption_result(config_type, decrypted_config):
    return f"Decrypted {config_type} Config:\n{decrypted_config}"

def replace_needless_characters(text):
    return text.replace("&", "\n").replace("?", "\n")

File: armod_decryptor.js
const crypto = require('crypto');
const base64 = require('base-64');
const { unescape } = require('querystring');

const ENCRYPTION_KEY = "YXJ0dW5uZWw3ODc5Nzg5eA==";

function decryptConfig(encryptedConfig) {
    const key = Buffer.from(ENCRYPTION_KEY, 'base64');
    const cipher = crypto.createDecipheriv('aes-128-ecb', key, null);
    const decrypted = Buffer.concat([cipher.update(Buffer.from(encryptedConfig, 'base64')), cipher.final()]);
    return decrypted.toString('utf-8');
}

function processConfig(config) {
    const patternVmess = /^ar-(vmess):\/\//;
    const patternSSH = /^ar-(ssh|vless|socks|trojan-go|trojan|ssr):\/\//;

    if (patternVmess.test(config)) {
        const encryptedConfig = config.replace(patternVmess, '');
        const decryptedConfig = decryptConfig(encryptedConfig);
        const decodedConfig = unescape(decryptedConfig);
        return formatDecryptionResult("VMess", decodedConfig);
    } else if (patternSSH.test(config)) {
        const encryptedConfig = config.replace(patternSSH, '');
        const decryptedConfig = decryptConfig(encryptedConfig);
        const decodedConfig = unescape(decryptedConfig);
        return formatDecryptionResult("SSH", decodedConfig.replace(/&/g, '\n').replace(/\?/g, '\n'));
    } else {
        return "Invalid config or unlock config";
    }
}

function formatDecryptionResult(configType, decryptedConfig) {
    return `Decrypted ${configType} Config:\n${decryptedConfig}`;
}

module.exports = { processConfig };
